### DISCUSSION TITLE
Improve the Editable Children functionality in the Scene dock

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#4092] [Editor]
The current situation with editable childre:

- The current usage is limited, it just allow to create variants of the original scene, any change you want to make permanent needs to be done out of the context of the instanced scene, which can be cumbersome and the ways to solve it currently are just workarounds (overwrite the original, or copy parameters, etc.) .

- One has to check every node to see the changes, there is no highlight on the nodes with modifications

![imagen](https://user-images.githubusercontent.com/10215987/155256417-72130972-949b-4531-9729-96d398bc407e.png)

- When closing the editable there is no way to know what changes are being discarded.
![imagen](https://user-images.githubusercontent.com/10215987/155256530-1ae49e26-afce-4b91-a019-683aacde88a7.png)

--------------------------------------

Improvements that can be done on this:

- Add an option to *apply changes* to the *base scene* (and *reset changes* can be useful too).
- Add an icon, or (because there are too many icons already) something like underscore to the names of the nodes with changes
- List the *changed properties* on the "close editable mode popup" (full path of changes)

**Comment by Zireael07** [comment]
+1 to highlighting nodes with modifications.

**Comment by azur-wolve** [comment]
What about mark the changed nodes with a red dot instead?

**Reply by eon-s** [reply]
yes, anything can work, it is getting a bit cluttered with icons, a dot like in some programs that indicate "new notification" on user icons (but in this case on the node icon) can be good too.

**Comment by Calinou** [comment]
> (and reset changes can be useful too).

This can be done by disabling Editable Children then enabling Editable Children again. We could have a menu action for doing this as a shortcut, but I'm not sure if it will be used often enough.

**Reply by eon-s** [reply]
Yeah, is some extra feature that can be easily added later if needed.
